azurelinuxagent/ga/exthandlers.py

```diff
@@ -486,33 +491,44 @@ def handle_enable(self, ext_handler_i):
             ext_handler_i.update_settings()
 
         ext_handler_i.enable()
+        # Remove the UNINSTALL_FAILED from env variables list after calling Install and Enable
+        ext_handler_i.remove_key_from_handler_env(UNINSTALL_FAILED)
```

Comment:
if enable throws the env doesn't get updated

Reply:
I'm setting the handler_env in the `ExtHandlerInstance` object. If enable throws then that particular object is deleted and no future states are affected.